src/main/java/htsjdk/samtools/cram/structure/SubstitutionMatrix.java

```diff
@@ -28,38 +33,29 @@
  * and an expanded in-memory form used for fast bi-directional interconversion between bases and
  * substitution codes during read and write.
  *
- * This implementation substitutes both upper and lower case versions of a give base with the
- * same (upper case) substitute base.
+ * This implementation allows both upper and lower case versions of a given reference base
+ * to be substituted with the same (upper case) substitute base (except for 'N', which is only
+ * handled for upper case) although it does not *generate* substitutions for lower case reference
+ * bases.
  */
-public class SubstitutionMatrix {
-    // List of bases subject to substitution. The upper and lower case arrays must have the bases
-    // in the same order so that the index for a given base is the same for upper and lower case
-    static final byte[] SUBSTITUTION_BASES_UPPER = new byte[]{'A', 'C', 'G', 'T', 'N'};
+ public class SubstitutionMatrix {
+    private static final Log log = Log.getInstance(Substitution.class);
+
+    // substitution bases, in the order in which they're stored in the substitution matrix
+    private static List<SubstitutionBase> BASES = Arrays.asList(SubstitutionBase.values());
```

Comment:
Is there a reason to not use `SubstitutionBase[] BASES` here? It looks like all uses of `BASES` are in `for` loops which would support arrays.

Reply:
@pshapiro4broad You're right that it could go either way based on the usages, and I don't mind changing it, but its is a static list - is there a compelling reason that array is better here ?

Reply:
It avoids the need to create an `Arrays.asList()` wrapper.